src/python/pants/option/native_options.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import inspect
+import logging
+import shlex
+from enum import Enum
+from typing import Any, Mapping, Optional, Sequence, Tuple
+
+from pants.engine.internals import native_engine
+from pants.engine.internals.native_engine import PyConfigSource
+from pants.option.config import ConfigSource
+from pants.option.custom_types import _flatten_shlexed_list, shell_str
+from pants.option.errors import OptionsError
+from pants.option.ranked_value import Rank
+from pants.util.strutil import get_strict_env
+
+logger = logging.getLogger()
+
+
+class NativeOptionParser:
+    """A Python wrapper around the Rust options parser."""
+
+    int_to_rank = [
+        Rank.NONE,
+        Rank.HARDCODED,
+        Rank.CONFIG_DEFAULT,
+        Rank.CONFIG,
+        Rank.ENVIRONMENT,
+        Rank.FLAG,
+    ]
+
+    def __init__(
+        self,
+        args: Optional[Sequence[str]],
+        env: Mapping[str, str],
+        config_sources: Optional[Sequence[ConfigSource]],
+        allow_pantsrc: bool,
+    ):
+        py_config_sources = (
+            None
+            if config_sources is None
+            else [PyConfigSource(cs.path, cs.content) for cs in config_sources]
+        )
+        self._native_parser = native_engine.PyOptionParser(
+            args,
+            dict(get_strict_env(env, logger)),
+            py_config_sources,
+            allow_pantsrc,
+        )
+
+        # (type, member_type) -> native get for that type.
+        self._getter_by_type = {
+            (bool, None): self._native_parser.get_bool,
+            (int, None): self._native_parser.get_int,
+            (float, None): self._native_parser.get_float,
+            (str, None): self._native_parser.get_string,
+            (list, bool): self._native_parser.get_bool_list,
+            (list, int): self._native_parser.get_int_list,
+            (list, float): self._native_parser.get_float_list,
+            (list, str): self._native_parser.get_string_list,
+            (dict, None): self._native_parser.get_dict,
+        }
+
+    def get(
+        self, *, scope, flags, default, option_type, member_type=None, passthrough=False
+    ) -> Tuple[Any, Rank]:
+        def is_enum(typ):
+            # TODO: When we switch to Python 3.11, use: return isinstance(typ, EnumType)
+            return inspect.isclass(typ) and issubclass(typ, Enum)
+
+        # '--foo.bar-baz' -> ['foo', 'bar', 'baz']
+        name_parts = flags[-1][2:].replace(".", "-").split("-")
+        switch = flags[0][1:] if len(flags) > 1 else None  # '-d' -> 'd'
+        option_id = native_engine.PyOptionId(*name_parts, scope=scope or "GLOBAL", switch=switch)
```

Comment:
I note that the `scope` arg seems to do this defaulting internally:  https://github.com/pantsbuild/pants/blob/d8bb287bd38edb0a843c080b4e0c5d5a8b3a2710/src/rust/engine/src/externs/options.rs#L110

```suggestion
        option_id = native_engine.PyOptionId(*name_parts, scope=scope, switch=switch)
```


Unless `scope` can be an empty string too?

Reply:
This is correct, `scope` cannot be an empty string

Reply:
Ah no, turns out this is necessary because on the Python side it's represented as an empty string, not None.

Reply:
That is, a scope can be an empty string, but that means GLOBAL.